modules/io/src/common/loaders.js

```diff
@@ -198,6 +199,78 @@ function getPBEXVIZType(arrayBuffer) {
   return envelope.type;
 }
 
+/* We need to modify the protobufjs objects to work closer to "normal"
+ * Javascript objects. The protobuf type reflection is available as `msg.$type`
+ * which is traversed in parallel to the `msg`.
+ *
+ * The following mutations are made:
+ *
+ * - Remove empty Arrays from the instance, protobuf.js will set to the default value: [] (the empty array)
+ * - Enum => String
+ * - Handle arrays of enum
+ * - Recursive on objects
+ */
+/* eslint-disable max-depth, complexity */
+function postProcessProtobuf(msg, pbType) {
```

Comment:
I think deserves a dedicated test function really.  With my experience in Protobuf.js with this you can easily miss some corner cases and a test makes it much easier to absorb new things.

Reply:
Yes, not that I have fixed the ui_config structure handling i'll will beef up code coverage